examples/extending/extending/agent_service.py

```diff
@@ -1,68 +1,51 @@
-# In addition, it's easy to scale up and down the agent service based on the workload. Agent services are designed to be language-agnostic.
-# For now, we only support Python agent, but we may support other languages in the future.
-#
-# Agent is designed to run a specific type of task. For example, you can create a BigQuery agent to run BigQuery task. Therefore, if you create a new type of task, you can
-# either run the task in the pod, or you can create a new agent to run it. You can determine how the task will be executed in the FlytePropeller configMap.
-#
-# Key goals of the agent service include:
-#
-# - Support for communication with external services: The focus is on enabling agents that seamlessly interact with external services.
-# - Independent testing and private deployment: Agents can be tested independently and deployed privately, providing flexibility and control over development.
-# - Flyte Agent usage in local development: Users, especially in flytekit and unionml, can leverage backend agents for local development, streamlining the development process.
-# - Language-agnostic: Agents can be authored in any programming language, allowing users to work with their preferred language and tools.
-# - Scalability: Agents are designed to be scalable, ensuring they can handle large-scale workloads effectively.
-# - Simple API: Agents offer a straightforward API, making integration and usage straightforward for developers.
-#
-# ## Why do we need an Agent Service?
-#
-# Without Agents, people need to implement a backend plugin in the propeller. The backend plugin is responsible for
-# creating a CRD and submitting a http request to the external service. However, it increases the complexity of flytepropeller, and
-# it's hard to maintain the backend plugin. For example, if we want to add a new plugin, we need to update and compile
-# flytepropeller, and it's also hard to test. In addition, the backend plugin is running in flytepropeller itself, so it
-# increases the load of the flytepropeller engine.
-#
-# Furthermore, implementing backend plugins can be challenging, particularly for data scientists and ML engineers who may lack proficiency in
-# Golang. Additionally, managing performance requirements, maintenance, and development can be burdensome.
-# To address these issues, we introduced the "Agent Service" in Flyte. This system enables rapid plugin
-# development while decoupling them from the core flytepropeller engine.
-#
-# ## Overview
-#
-# The Flyte Agent Service is a Python-based agent registry powered by a gRPC server. It allows users and flytepropeller
-# to send gRPC requests to the registry for executing jobs such as BigQuery and Databricks. Each Agent service is a Kubernetes
-# deployment. You can create two different Agent services hosting different Agents. For example, you can create one production
+# The Flyte Agent service is a Python-based agent registry powered by a gRPC server. It allows users and FlytePropeller
```

Comment:
do we need to uppercase the first letter in "agent" everywhere? i don't think so, right?

Reply:
It depends on how we want to brand agents / the agent service -- whatever the decision, I can make the docs consistent.

Reply:
okay, we can finalize this later.